expr/functions/aliasQuery/function.go

```diff
@@ -0,0 +1,138 @@
+package aliasQuery
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/go-graphite/carbonapi/expr/helper"
+	"github.com/go-graphite/carbonapi/expr/interfaces"
+	"github.com/go-graphite/carbonapi/expr/types"
+	"github.com/go-graphite/carbonapi/pkg/parser"
+)
+
+type aliasQuery struct {
+	interfaces.FunctionBase
+}
+
+func GetOrder() interfaces.Order {
+	return interfaces.Any
+}
+
+func New(_ string) []interfaces.FunctionMetadata {
+	return []interfaces.FunctionMetadata{
+		{Name: "aliasQuery", F: &aliasQuery{}},
+	}
+}
+
+func (f *aliasQuery) Do(ctx context.Context, e parser.Expr, from, until int64, values map[parser.MetricRequest][]*types.MetricData) ([]*types.MetricData, error) {
+	seriesList, err := helper.GetSeriesArg(ctx, e.Arg(0), from, until, values)
+	if err != nil {
+		return nil, err
+	}
+	search, err := e.GetStringArg(1)
+	if err != nil {
+		return nil, err
+	}
+	replace, err := e.GetStringArg(2)
+	if err != nil {
+		return nil, err
+	}
+	newName, err := e.GetStringArg(3)
+	if err != nil {
+		return nil, err
+	}
+
+	re, err := regexp.Compile(search)
+	if err != nil {
+		return nil, err
+	}
+	replace = helper.Backref.ReplaceAllString(replace, "$${$1}")
+
+	fetchTargets := make([]parser.Expr, len(seriesList))
+	for i, series := range seriesList {
+		newTarget := re.ReplaceAllString(series.Name, replace)
+		expr, _, err := parser.ParseExpr(newTarget)
+		if err != nil {
+			return nil, err
+		}
+		fetchTargets[i] = expr
+	}
+	targetValues, err := f.GetEvaluator().Fetch(ctx, fetchTargets, from, until, values)
+
+	results := make([]*types.MetricData, len(seriesList))
+
+	for i, series := range seriesList {
+		v, err := f.getLastValueOfSeries(ctx, fetchTargets[i], from, until, targetValues)
+		if err != nil {
+			return nil, err
+		}
+
+		n := fmt.Sprintf(newName, v)
+
+		var r *types.MetricData
+		if series.Name == n {
+			r = series.CopyLinkTags()
+			r.Tags["name"] = r.Name
+		} else {
+			r = series.CopyName(n)
+		}
+
+		results[i] = r
+	}
+
+	return results, nil
+}
+
+func (f *aliasQuery) getLastValueOfSeries(ctx context.Context, e parser.Expr, from, until int64, targetValues map[parser.MetricRequest][]*types.MetricData) (float64, error) {
+	res, err := f.GetEvaluator().Eval(ctx, e, from, until, targetValues)
```

Comment:
nitpicking: Would `GetSeriesArg` work here?

```suggestion
	res, err := helper.GetSeriesArg(ctx, e, from, until, targetValues)
```

Reply:
Good call, and it has that extra check in case something went horribly wrong building the expression.